File: app/api/src/Astra.Api/Llm/Archetypes/angular-dotnet8/canonical-delphi-angular-widget-crud/backend/Services/WidgetService.cs
using Demo.WidgetApi;
using Demo.WidgetApi.Models;

namespace Demo.WidgetApi.Services;

/// <summary>
/// Business rules for widgets, independent of both the HTTP layer
/// (<c>WidgetsController</c>) and the storage layer (<c>IWidgetRecordPort</c>).
/// </summary>
[SpecClaim("EC-1")]
public sealed class WidgetService(IWidgetRecordPort recordPort)
{
    public IReadOnlyList<Widget> List() => recordPort.List();

    /// <summary>EC-1: a blank name is rejected rather than silently stored.</summary>
    [SpecClaim("EC-1")]
    public Widget Create(NewWidget request)
    {
        var name = request.Name.Trim();
        if (name.Length == 0)
            throw new ArgumentException("Widget name is required.", nameof(request));

        return recordPort.Add(name, request.Quantity);
    }

    public bool Delete(int id) => recordPort.Remove(id);
}

File: app/api/src/Astra.Api/Llm/Archetypes/angular-java/canonical-cpp-widget-crud/backend/src/main/java/com/example/widget/WidgetService.java
package com.example.widget;

import java.util.List;
import org.springframework.stereotype.Service;

/**
 * Business rules for widgets, independent of both the HTTP layer
 * ({@link WidgetController}) and the storage layer ({@link WidgetRecordPort}).
 */
@Service
@SpecClaim("EC-1")
public class WidgetService {
    private final WidgetRecordPort recordPort;

    public WidgetService(WidgetRecordPort recordPort) {
        this.recordPort = recordPort;
    }

    public List<Widget> list() {
        return recordPort.list();
    }

    /** EC-1: a blank name is rejected rather than silently stored. */
    @SpecClaim("EC-1")
    public Widget create(NewWidgetRequest request) {
        var name = request.name().trim();
        if (name.isEmpty()) {
            throw new IllegalArgumentException("Widget name is required.");
        }
        return recordPort.add(name, request.quantity());
    }

    public boolean delete(int id) {
        return recordPort.remove(id);
    }
}
